examples/debug.rs

```diff
@@ -0,0 +1,210 @@
+    tags: Vec<(&'ar str, &'ar str)>,
+}
+
+#[derive(Debug)]
+struct Way<'ar> {
+    id: i64,
+    tags: Vec<(&'ar str, &'ar str)>,
+    nodes: Vec<u64>,
+}
+
+#[derive(Debug)]
+struct Relation<'ar> {
+    id: i64,
+    tags: Vec<(&'ar str, &'ar str)>,
+    members: Vec<Member<'ar>>,
+}
+
+#[derive(Debug)]
+struct Member<'ar> {
+    type_: Type,
+    idx: u64,
+    role: &'ar str,
+}
+
+#[derive(Debug)]
+enum Type {
+    Node,
+    Way,
+    Relation,
+}
+
+impl<'ar> Member<'ar> {
+    fn new_slice(
+        archive: &'ar Osm,
+        relation_idx: usize,
+    ) -> impl Iterator<Item = Result<Member<'ar>, Utf8Error>> {
+        let strings = archive.stringtable();
+        archive
+            .relation_members()
+            .at(relation_idx as usize)
+            .map(move |member| {
+                let res = match member {
+                    RefRelationMembers::NodeMember(m) => Member {
+                        type_: Type::Node,
+                        idx: m.node_idx(),
+                        role: strings.substring(m.role_idx() as usize)?,
+                    },
+                    RefRelationMembers::WayMember(m) => Member {
+                        type_: Type::Way,
+                        idx: m.way_idx(),
+                        role: strings.substring(m.role_idx() as usize)?,
+                    },
+                    RefRelationMembers::RelationMember(m) => Member {
+                        type_: Type::Relation,
+                        idx: m.relation_idx(),
+                        role: strings.substring(m.role_idx() as usize)?,
+                    },
+                };
+                Ok(res)
+            })
+    }
+}
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let mut args = std::env::args().skip(1).take(2);
+    let archive_dir = args.next().ok_or(
+        "USAGE: debug <osmflat-archive> [TYPES] \
+         TYPES can be any combination of 'n', 'w', 'r' (default: 'nwr').",
+    )?;
+    let types = args.next().unwrap_or("nrw".to_string());
+    let archive = Osm::open(FileResourceStorage::new(archive_dir))?;
+
+    println!("{:#?}", Header::new(&archive)?);
+
+    if types.contains('n') {
+        for node in archive.nodes().slice(..3) {
+            let tags: Result<Vec<_>, _> = osmflat::tags(&archive, node.tags()).collect();
+            let node = Node {
+                id: node.id(),
+                lat: FixedI64(node.lat()),
+                lon: FixedI64(node.lon()),
+                tags: tags?,
+            };
+
+            println!("{:#?}", node);
+        }
+    }
+
+    let nodes_index = archive.nodes_index();
+
+    if types.contains('w') {
+        for way in archive.ways() {
+            let tags: Result<Vec<_>, _> = osmflat::tags(&archive, way.tags()).collect();
+            let way = Way {
+                id: way.id(),
+                tags: tags?,
+                nodes: way
+                    .refs()
+                    .map(|idx| nodes_index.at(idx as usize).value())
```

Comment:
nodes_index.slice(way.refs().start as usize..way.refs().end as usize)?

PS: We could enhance flatdata to allow indexing by non-usize Ranges

Reply:
Would still need to call `map` to get the value.